src/asm/fstack.cpp

```diff
@@ -198,15 +219,15 @@ bool yywrap() {
 		// If this wasn't the last iteration, wrap instead of popping
 		if (fileInfoIters.front() <= context.nbReptIters) {
 			lexer_RestartRept(context.fileInfo->lineNo);
-			context.uniqueID = macro_UseNewUniqueID();
+			context.uniqueIDStr->clear(); // Invalidate the current unique ID (if any).
 			return false;
 		}
 	} else if (contextStack.size() == 1) {
 		return true;
 	}
 
-	Context oldContext = contextStack.top();
-
+	// TODO: can't we just perform all ops that require `oldContext` *before* `pop()`?
```

Comment:
Yeah, but can't *that* be done after the `pop()`, and the rest *before*?

Reply:
Sure, but is there any reason to, if we can't eliminate the `oldContext` local by doing so?

Like, we could do this:

```diff
-if (Context &context = contextStack.top(); context.fileInfo->type == NODE_REPT) {
+Context &context = contextStack.top();
+
+if (context.fileInfo->type == NODE_REPT) {
 	...
 } else if (contextStack.size() == 1) {
 	return true;
 }

+	// Free the file stack node
+	if (!context.fileInfo->referenced)
+		delete context.fileInfo;
+
 	Context oldContext = std::move(contextStack.top());
 	contextStack.pop();

 	lexer_CleanupState(oldContext.lexerState);
 	// Restore args if a macro (not REPT) saved them
 	if (oldContext.fileInfo->type == NODE_MACRO)
 		macro_UseNewArgs(contextStack.top().macroArgs);
-	// Free the file stack node
-	if (!oldContext.fileInfo->referenced)
-		delete oldContext.fileInfo;

 	lexer_SetState(&contextStack.top().lexerState);
```

But I don't think it would improve clarity of what's being done.

Reply:
Actually, #1371 and the macro-arg changes I added here have now made `oldContext` unnecessary after all. ^^